students/rick_halsell/lesson03/list_lab.py

```diff
@@ -0,0 +1,117 @@
+# Prompts user to add another fruit to the list, but at the beginning of the list
+FruitResponseBegin = input('3. Please enter the name of a fruit to add to the beginning of the list > ')
+# adding fruit to the beginning of list using +
+fruit = [FruitResponseBegin] + fruit
+# printing list contents after addition
+print(fruit)
+# Prompts user to add another fruit to the beginning of the list
+FruitResponseBeginTwo = input('4. Please enter the another name of a fruit to add to the beginning of the list > ')
+# # adding an additional item to the beginning of the list using insert()
+fruit.insert(0, FruitResponseBeginTwo)
+# printing list contents after second addition
+print(fruit)
+# first sort the list and make uppercase
+fruit.sort(key=str.upper)
+# Display all the fruits that begin with “P”, using a for loop
+print('All items that begin With the letter P')
+for item in fruit:
+    item = item.capitalize() # capitalized the first letter of each item
+    if item.startswith('P'): # searched for all items that start with 'P'
+        print(item) # printed the item
+# Series 2
+# Display the list
+print(fruit)
+#Remove the last fruit from the list.
+del fruit[-1]
+# Display the list after last item removed
+print(fruit)
+# Ask the user for a fruit to delete, find it and delete it.
+fruit = fruit * 2
+print(f'The list has been doubled: \n {fruit}')
+while True:
+    FruitResponseDeletion = input('5. Please enter the name of a fruit to delete from the list > ')
+    if FruitResponseDeletion in fruit:
+        for item in fruit:
+            with contextlib.suppress(ValueError): # handle error when loop removes all occurrences
+                fruit.remove(FruitResponseDeletion)
+            continue
+        print(fruit)
+        break
+    else:
+        print('Fruit Not Found.')
+# Tells users which fruit was removed
+print(f'{FruitResponseDeletion} removed from the list.')
+# prints list again
+print(fruit)
+# (Bonus: Multiply the list times two. Keep asking until a match is found. Once found, delete all occurrences.)
+
+# Series 3
+# Ask the user for input displaying a line like “Do you like apples?”
+#for each fruit in the list (making the fruit all lowercase).
+for item in fruit:
+    item.lower() # lowers the case
+    answer = input(f'Y or N - Do you like {item}? ') # prompts the user
+    i = 0 # sets a value for the while loop
+    while i < len(fruit): # sets the conditions of the loop
+        if answer.lower() == 'y':
+            i += 1 # increments the value
+            continue
+        elif answer.lower() == 'n':
+            with contextlib.suppress(ValueError): # handle error when loop removes all occurrences
+                fruit.remove(item) # removes the value from the list
+                print(fruit) # prints the list after each iteration
+            i += 1 # increments the value
+            continue
+        else:
+            print('Please enter Y or N.')
+            continue
+print(fruit) # prints the final list
```

Comment:
When answering 'N' this is skipping the next item in the list - do you know why?